authd/authd.c

```diff
@@ -199,7 +199,7 @@ main(int argc, char *argv[])
 	rb_set_time();
 	setup_signals();
 
-	authd_option_handlers = rb_dictionary_create("authd options handlers", rb_strcasecmp);
+	authd_option_handlers = rb_dictionary_create("authd options handlers", (DCF)rb_strcasecmp);
```

Comment:
Casting function pointers depends on the representation of `const char *` and `const void *` being the same, which is likely but not guaranteed. It would be safer to have wrappers (one for `rb_strcasecmp` and one for `irccmp`) that can be passed to `rb_dictionary_create` without a conversion.

Since `rb_strcasecmp` is part of librb, librb could also provide this wrapper. The wrapper for `irccmp` will need to be in ircd/.

By the way, the unsafety is similar to the previous unsafety since the code in librb/src/dictionary.c calls the unprototyped function with two `const void *` in all cases. The change to a prototyped function adds checking to those calls.